hydromt/models/model_api.py

```diff
@@ -211,6 +218,88 @@ def update(self, model_out=None, write=True, opt=None):
         if write:
             self.write()
 
+    ## general setup methods
+
+    def setup_region(
```

Comment:
Super nice! Should replace the equivalent use in the different plugins (with a check on kind afterwards)

Reply:
yes! For hydromt_sfincs I'm already using it even without wrapping. For hydromt_wflow we could wrap it in the setup_basemaps function or split it